local_pathfinding/node_navigate.py

```diff
@@ -176,7 +178,22 @@ def get_desired_heading(self) -> float:
         )
 
         # TODO: create function to compute the heading from current position to next local waypoint
-        return 0.0
+        # Find index of self.gps in current waypoints
+        current_waypoint = self.gps.lat_lon
+        next_waypoint = None
+        waypoints = self.local_path.waypoints
+        for i in range(len(waypoints)):
+            if waypoints[i] == current_waypoint and i != len(waypoints) - 1:
+                next_waypoint = waypoints[i+1]
+                break
+        if next_waypoint is None:
+            return 0.0
+        heading = math.atan2(
+            (next_waypoint.longitude-current_waypoint.longitude) *
+            math.cos(current_waypoint.latitude),
+            (next_waypoint.latitude-current_waypoint.latitude)
+        )
+        return heading
```

Comment:
Would it be possible to add some tests for this function?

Reply:
For the tests, should I create a new test_node_navigate.py file and manually set the value of self.gps to calculate the desired heading?

Reply:
I think this would be the best way to test this:
1. Create a helper function to `get_desired_heading()` that accepts "simple" arguments like waypoints or lists of waypoints, rather than "complicated" arguments like `self` (which is a whole sailbot object). `get_desired_heading()` will basically just call the helper function.
2. Create `test_node_navigate.py` like you suggested, and just test the helper function. I'd suggest using simple tests like when the desired heading is $0 \degree$ or $90 \degree$, and maybe some more complicated tests where the "actual" value is computed with geodesic.inv() as well.